synapse/replication/http/federation.py

```diff
@@ -69,14 +69,37 @@ def __init__(self, hs: "HomeServer"):
         self.federation_event_handler = hs.get_federation_event_handler()
 
     @staticmethod
-    async def _serialize_payload(store, room_id, event_and_contexts, backfilled):
+    async def _serialize_payload(
+        store,
+        room_id,
+        event_and_contexts,
+        inhibit_push_notifications,
+        should_calculate_state_and_forward_extrems,
+        use_negative_stream_ordering,
+        inhibit_local_membership_updates,
+        update_room_forward_stream_ordering,
```

Comment:
Is there a way we can define the arguments here like the following? I can't get it to work even without the keyword-only marker.

```py
async def _serialize_payload(
    store,
    room_id,
    event_and_contexts,
    *,
    inhibit_push_notifications: bool = False,
    should_calculate_state_and_forward_extrems: bool = True,
    use_negative_stream_ordering: bool = False,
    inhibit_local_membership_updates: bool = False,
    update_room_forward_stream_ordering: bool = True,
):
```
    
When I do this, I get the following error:

```
synapse/replication/http/federation.py:72: error: Signature of "_serialize_payload" incompatible with supertype "ReplicationEndpoint"  [override]
```

And I can't figure out how to adjust the abstract method to be more flexible, https://github.com/matrix-org/synapse/blob/7ffddd819c2e0e7edff141e987372205894084b6/synapse/replication/http/_base.py#L131-L132

Reply:
I'm not entirely sure why, but it's the `:bool` type annotations that are upsetting it rather than the `*` or the default values. If I remove the `:bool`s, it's fine.